source/scripts/app.js

```diff
@@ -502,10 +502,23 @@ function homeSearchFunction() {
     if (e.key === "Enter") {
       e.preventDefault();
       // store search string and navigate to explore page
-      // let searchQuery = e.target.value;
+      let searchQuery = { query: e.target.value };
+
+      //clear search and route to explore
+      e.target.value = "";
       router.navigate("explore-page");
 
-      // TODO more here once explore page setup
+      //display results of search
+      let explore = document.querySelector("explore-page");
+      let shadow = explore.shadowRoot;
+      let bar = shadow.getElementById("search-bar");
+      bar.value = searchQuery.query;
+
+      if (searchQuery.query.length) {
+        populateExplorePage(searchQuery);
+      } else {
+        populateExplorePage();
+      }
```

Comment:
I forgot, we also need to toggle the type of the explore page. This allows the explore page to behave differently when it's in explore (random recipes) mode or in search (not random recipes) mode
```suggestion
      //display results of search
      let explore = document.querySelector("explore-page");
      let shadow = explore.shadowRoot;
      let topLevel = shadow.getElementById("explore-top-level");
      let bar = shadow.getElementById("search-bar");
      bar.value = searchQuery.query;

      if (searchQuery.query.length) {
        if (topLevel.classList.contains("type-explore")) {
          toggleExplorePageType();
        }
        populateExplorePage(searchQuery);
      } else {
        if (!topLevel.classList.contains("type-explore")) {
          toggleExplorePageType();
        }
        populateExplorePage();
      }
```